### PR TITLE
Add different map types.

### PR DESCRIPTION
**Pull request by kpancha**
Week 4 walkthrough: maps https://github.com/kpancha/step/blob/master/walkthroughs/week-4-libraries/maps/maps-walkthrough.md#customizing-your-map

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -192,7 +192,42 @@ function createNextButton(direction, isValid) {
...
+      {"featureType": "road.highway.controlled_access","elementType": "geometry","stylers": [{"color": "#e98d58"}]},
+      {"featureType": "road.highway.controlled_access","elementType": "geometry.stroke","stylers": [{"color": "#db8555"}]},
+      {"featureType": "road.local","elementType": "labels.text.fill","stylers": [{"color": "#806b63"}]},
+      {"featureType": "transit.line","elementType": "geometry","stylers": [{"color": "#dfd2ae"}]},
+      {"featureType": "transit.line","elementType": "labels.text.fill","stylers": [{"color": "#8f7d77"}]},
+      {"featureType": "transit.line","elementType": "labels.text.stroke","stylers": [{"color": "#ebe3cd"}]},
+      {"featureType": "transit.station","elementType": "geometry","stylers": [{"color": "#dfd2ae"}]},
+      {"featureType": "water","elementType": "geometry.fill","stylers": [{"color": "#b9d3c2"}]},
+      {"featureType": "water","elementType": "labels.text.fill","stylers": [{"color": "#92998d"}]}],
+      {name: 'Retro Map'});
```
Comment (kpancha) [inline portfolio/src/main/webapp/script.js:220]: This was how the demo says to do styling but I don't think what I have here is very readable

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -230,4 +230,43 @@ function createMap() {
...
   map.mapTypes.set('styled_map', styledMapType);
   map.setMapTypeId('styled_map');
+
+  const locationMarkers = getMarkers();
+  for (let marker of locationMarkers) {
+    marker.setMap(map);
+  }
+}
+
+function getMarkers() {
```
Comment (kpancha) [inline portfolio/src/main/webapp/script.js:240]: Comment this function

**Comment by kpancha** [comment]
I am thinking about how to divide creating and setting markers so it doesn't turn into one long function. This is what I came up with but it still seems like the functions are pretty long. Let me know if it looks like I'm on the right track or if I could be doing something differently for this

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -267,6 +258,7 @@ function setMarkers(map, markers, icon=null) {
  * Creates a marker witha given google.maps.LatLng position and title.
  */
 function createMarker(position, title) {
+
```
Comment (kpancha) [inline portfolio/src/main/webapp/script.js:261]: remove blank line

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -191,9 +191,196 @@ function createNextButton(direction, isValid) {
...
+  return new google.maps.Marker({position, title});
+}
+
+/**
+ * Returns an array of markers for places lived.
+ */
+function getPlacesLivedMarkers() {
+  const placesLivedMarkers = [];
+
+  const montyLatlng = new google.maps.LatLng({lat: 40.44, lng: -74.66});
```
Comment (jesseelizabeth) [inline portfolio/src/main/webapp/script.js:271]: nit: also create a createLatLng helper like you did for createMarker

**Pull request merged**